src/components/Table/Table.tsx

```diff
@@ -75,6 +77,20 @@ export interface ITableProps {
    * Loading state handled by Suspense in parent component.
    */
   emptyState?: React.ReactNode;
+  /**
+   * If defined, it will show a checkbox to select rows. The
+   * state is to be maintained by the parent component.
+   *
+   * Usage:
+   *
+   * const [selectedRows, setSelectedRows] = useState<RowSelectionState>({});
+   * const selectedRowsProp = useMemo(() => ({state: selectedRows, setState: setSelectedRows}), [selectedRows, setSelectedRows])
+   * <Table selectedRows={selectedRowsProp} />
+   */
+  selectedRows?: {
```

Comment:
I have added this prop because we need the state outside of `<Table />` so that it can be passed to `<TableToolbar />`. The other option was to pull the react-table upto `<TablePage />`.